src/mainboard/protectli/vault_adl_p/Kconfig

```diff
@@ -3,17 +3,19 @@ if BOARD_PROTECTLI_VP66XX
 config BOARD_SPECIFIC_OPTIONS
 	def_bool y
 	select BOARD_ROMSIZE_KB_16384
-	select SOC_INTEL_ALDERLAKE_PCH_P
-	select SOC_INTEL_COMMON_BLOCK_HDA_VERB
-	select SOC_INTEL_ENABLE_USB4_PCIE_RESOURCES
-	select SUPERIO_ITE_IT8659E
-	select DRIVERS_UART_8250IO
 	select DRIVERS_I2C_GENERIC
 	select DRIVERS_INTEL_PMC
+	select DRIVERS_UART_8250IO
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
 	select INTEL_GMA_HAVE_VBT
 	select MEMORY_MAPPED_TPM
+	select SOC_INTEL_ALDERLAKE_PCH_P
+	select SOC_INTEL_COMMON_BLOCK_HDA_VERB
+	select SOC_INTEL_COMMON_OC_WDT_ENABLE
+	select SOC_INTEL_COMMON_OC_WDT_WDAT
+	select SOC_INTEL_ENABLE_USB4_PCIE_RESOURCES
+	select SUPERIO_ITE_IT8659E
```

Comment:
Why do we have to change the order of options here? SOC_INTEL_ALDERLAKE_PCH_P is first for a reason, it selects most of the microarchitecture specific options.

`SOC_INTEL_COMMON_OC_WDT_ENABLE` this should not be selected. It means we cannot disable the watchdog.
`SOC_INTEL_COMMON_OC_WDT` is selected by ADL_P already.

Reply:
Just alphabetized them. Will revert.

Reply:
https://github.com/Dasharo/coreboot/pull/472/commits/6206afd8b85a7656a109566cc1e5f6d2c39e9674

Reply:
>  SOC_INTEL_COMMON_OC_WDT is selected by ADL_P already.

Here I'm selecting ` SOC_INTEL_COMMON_OC_WDT_WDAT` (maybe the naming is unfortunate)

Reply:
> SOC_INTEL_COMMON_OC_WDT_ENABLE this should not be selected. It means we cannot disable the watchdog.

@miczyg1 are you sure about that?

https://github.com/Dasharo/coreboot/blob/6b4a0b81635b919c6fa21814b3b95c7612fe5a6f/src/soc/intel/common/block/oc_wdt/oc_wdt.c#L95
https://github.com/Dasharo/coreboot/blob/6b4a0b81635b919c6fa21814b3b95c7612fe5a6f/payloads/external/edk2/Makefile#L248-L251

Reply:
What I mean is `select SOC_INTEL_COMMON_OC_WDT_ENABLE` will prevent you from chanign this option in the menuconfig (it will be forcibly selected as enabled and attempts to disable it will not work)

Reply:
An acceptable solution would be

```
config SOC_INTEL_COMMON_OC_WDT_ENABLE
       default y
```

But thas shadows the options being set in configs/config.protectli* files, so you cannot early determine what is enabled and what not, unless you look deeper into the code, which I am not in favor of...

Reply:
Oh I see what you mean. Will fix

Reply:
Fixed in https://github.com/Dasharo/coreboot/pull/472/commits/6df14ccf6f0782300edb8a057b2067088193d15f